components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ApplicationThrottleController.java

```diff
@@ -62,7 +67,26 @@ public static ThrottleContext getApplicationThrottleContext(MessageContext synCt
     private static ThrottleContext createThrottleContext(MessageContext synCtx, ThrottleDataHolder dataHolder, String applicationId){
 
         //Object entryValue = synCtx.getEntry(APPLICATION_THROTTLE_POLICY_KEY);
-        Object entryValue = lookup(APPLICATION_THROTTLE_POLICY_KEY);
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        RealmService realmService = (RealmService) carbonContext.getOSGiService(RealmService.class, null);
+
+        AuthenticationContext authContext = APISecurityUtils.getAuthenticationContext(synCtx);
+        //extract the subscriber username from the auth Context
+        String subscriber = authContext.getSubscriber();
+
+        //get the tenant Domain from the subscriber
+        String tenantDomain = MultitenantUtils.getTenantDomain(subscriber);
+        int tenantId;
+
+        //get the tenant domain id from the tenant domain name
+        try {
+            tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
+        } catch (UserStoreException e) {
+            handleException("Unable to Find the tenant ID using tenant: " + tenantDomain);
```

Comment:
yes. i have missed it. it is fixed in commit: https://github.com/wso2/carbon-apimgt/commit/e433e625a5bed755ccab07851ac9510361b9e400